jOOQ/src/main/java/org/jooq/Name.java

```diff
@@ -435,6 +445,270 @@
     @Support({ FIREBIRD, H2, HSQLDB, POSTGRES })
     DerivedColumnList22 fields(String fieldName1, String fieldName2, String fieldName3, String fieldName4, String fieldName5, String fieldName6, String fieldName7, String fieldName8, String fieldName9, String fieldName10, String fieldName11, String fieldName12, String fieldName13, String fieldName14, String fieldName15, String fieldName16, String fieldName17, String fieldName18, String fieldName19, String fieldName20, String fieldName21, String fieldName22);
 
+    /**
+     * Add a list of fields to this name to make this name a
+     * {@link DerivedColumnList}.
+     * <p>
+     * The <code>DerivedColumnList</code> can then be used along with a
+     * subselect to form a {@link CommonTableExpression} to be used with
+     * <code>WITH</code> clauses.
+     */
+    @Generated("This method was generated using jOOQ-tools")
+    @Support({ FIREBIRD, H2, HSQLDB, POSTGRES })
+    DerivedColumnList1 fields(Name fieldName1);
+
+    /**
+     * Add a list of fields to this name to make this name a
+     * {@link DerivedColumnList}.
+     * <p>
+     * The <code>DerivedColumnList</code> can then be used along with a
+     * subselect to form a {@link CommonTableExpression} to be used with
+     * <code>WITH</code> clauses.
+     */
+    @Generated("This method was generated using jOOQ-tools")
+    @Support({ FIREBIRD, H2, HSQLDB, POSTGRES })
+    DerivedColumnList2 fields(Name fieldName1, Name fieldName2);
```

Comment:
Oh, I'm sorry you went through all this trouble :) We have a code generator (not open source), which generates this degree-22 API madness for us...